src/Executor/Executor.php

```diff
@@ -159,13 +161,18 @@ public function createExecution(
             'timeout' => $timeout,
         ];
 
-        $response = $this->call(self::METHOD_POST, $route, $headers, $params, true, 30);
+        /* Add 2 seconds as a buffer to the actual timeout value since there can be a slight variance*/
+        $requestTimeout  = $timeout + 2;
+
+        $response = $this->call(self::METHOD_POST, $route, $headers, $params, true, $requestTimeout);
         $status = $response['headers']['status-code'];
 
-        if ($status >= 400) {
-            for ($attempts = 0; $attempts < 10; $attempts++) {
-                switch ($status) {
-                    case 404:
+        for ($attempts = 0; $attempts < 10; $attempts++) {
+            try {
+                switch (true) {
+                    case $status < 400:
+                        return $response['body'];
+                    case $status == 404:
```

Comment:
I would suggest using strict comparison here with `===` to prevent type juggling.

Reply:
addressed 👍